src/test/java/com/google/cloud/spanner/pgadapter/PgAdapterTestEnv.java

```diff
@@ -184,8 +182,19 @@ public String getDatabaseId() {
     if (databaseId == null) {
       databaseId = System.getProperty(TEST_DATABASE_PROPERTY, DEFAULT_DATABASE_ID);
     }
-    return String.format(
-        "%s_%d_%d", databaseId, System.currentTimeMillis(), new Random().nextInt(Short.MAX_VALUE));
+    String id =
+        String.format(
+            "%s_%d_%d",
+            databaseId, System.currentTimeMillis(), new Random().nextInt(Short.MAX_VALUE));
+    // Make sure the database id is not longer than the max allowed 30 characters.
+    if (id.length() > 30) {
```

Comment:
The randomly chosen database ID could be invalid depending on the length of the configured prefix. If the prefix was long, the addition of a random number could push the total length of the id beyond the allowed max of 30 characters.